BUG: stats: Two new XSLOW test failures

There are two new XSLOW test failures occurring in `stats`:

```
$ SCIPY_XSLOW=1 python3 dev.py test -t scipy.stats.tests.test_stats -m full -j 9
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[2/2] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.14.0.dev0+1096.9c75b1f, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
=============================== test session starts ================================
platform linux -- Python 3.12.3, pytest-8.1.1, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: timeout-2.3.1, xdist-3.5.0, hypothesis-6.100.1, cov-5.0.0
9 workers [1019 items]  
............................................................................ [  7%]
............................................................................ [ 14%]
............................................................................ [ 22%]
............................................................................ [ 29%]
............................................................................ [ 37%]
............................................................................ [ 44%]
............................................................................ [ 52%]
........................................x................................... [ 59%]
...........................................x.......................x........ [ 67%]
............................................................................ [ 74%]
............................................................................ [ 82%]
............................................................................ [ 89%]
...............................................s............................ [ 96%]
.FF............................                                              [100%]
===================================== FAILURES =====================================
____________________ TestStudentTest.test_pvalue_ci[less-numpy] ____________________
[gw0] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/stats/tests/test_stats.py:3708: in test_pvalue_ci
    @hypothesis.given(alpha=hypothesis.strategies.floats(1e-15, 1-1e-15),
        alternative = 'less'
        f          = <function given.<locals>.run_test_as_given.<locals>.wrapped_test at 0x710e04b10360>
        self       = <scipy.stats.tests.test_stats.TestStudentTest object at 0x710dfc1869c0>
        xp         = <module 'numpy' from '/home/warren/py3.12.3/lib/python3.12/site-packages/numpy/__init__.py'>
scipy/stats/tests/test_stats.py:3714: in test_pvalue_ci
    data = data.astype(copy=True)  # ensure byte order
E   TypeError: astype() missing required argument 'dtype' (pos 0)
E   Falsifying example: test_pvalue_ci(
E       # The test always failed when commented parts were varied together.
E       self=<scipy.stats.tests.test_stats.TestStudentTest object at 0x710dfc1869c0>,
E       alternative='less',
E       xp=<module 'numpy' from '/home/warren/py3.12.3/lib/python3.12/site-packages/numpy/__init__.py'>,
E       alpha=0.5,  # or any other generated value
E       data_axis=(array([0., 1.], dtype=float16),
E        0),  # or any other generated value
E   )
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.100.1', b'AXicY2BgOMAAA4xghBUwAgAc3wDE') as a decorator on your test case
        alpha      = 0.5
        alternative = 'less'
        axis       = 0
        data       = array([0., 1.], dtype=float16)
        data_axis  = (array([0., 1.], dtype=float16), 0)
        self       = <scipy.stats.tests.test_stats.TestStudentTest object at 0x710dfc1869c0>
        xp         = <module 'numpy' from '/home/warren/py3.12.3/lib/python3.12/site-packages/numpy/__init__.py'>
__________________ TestStudentTest.test_pvalue_ci[greater-numpy] ___________________
[gw0] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/stats/tests/test_stats.py:3708: in test_pvalue_ci
    @hypothesis.given(alpha=hypothesis.strategies.floats(1e-15, 1-1e-15),
        alternative = 'greater'
        f          = <function given.<locals>.run_test_as_given.<locals>.wrapped_test at 0x710e04b10360>
        self       = <scipy.stats.tests.test_stats.TestStudentTest object at 0x710dfc186960>
        xp         = <module 'numpy' from '/home/warren/py3.12.3/lib/python3.12/site-packages/numpy/__init__.py'>
scipy/stats/tests/test_stats.py:3714: in test_pvalue_ci
    data = data.astype(copy=True)  # ensure byte order
E   TypeError: astype() missing required argument 'dtype' (pos 0)
E   Falsifying example: test_pvalue_ci(
E       self=<scipy.stats.tests.test_stats.TestStudentTest object at 0x710dfc186960>,
E       alternative='greater',
E       xp=<module 'numpy' from '/home/warren/py3.12.3/lib/python3.12/site-packages/numpy/__init__.py'>,
E       alpha=0.5,  # or any other generated value
E       data_axis=(array([0., 1.], dtype=float16), 0),
E   )
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.100.1', b'AXicY2BgOMAAA4xghBUwAgAc3wDE') as a decorator on your test case
        alpha      = 0.5
        alternative = 'greater'
        axis       = 0
        data       = array([0., 1.], dtype=float16)
        data_axis  = (array([0., 1.], dtype=float16), 0)
        self       = <scipy.stats.tests.test_stats.TestStudentTest object at 0x710dfc186960>
        xp         = <module 'numpy' from '/home/warren/py3.12.3/lib/python3.12/site-packages/numpy/__init__.py'>
============================= short test summary info ==============================
FAILED scipy/stats/tests/test_stats.py::TestStudentTest::test_pvalue_ci[less-numpy] - TypeError: astype() missing required argument 'dtype' (pos 0)
FAILED scipy/stats/tests/test_stats.py::TestStudentTest::test_pvalue_ci[greater-numpy] - TypeError: astype() missing required argument 'dtype' (pos 0)
========= 2 failed, 1013 passed, 1 skipped, 3 xfailed in 78.32s (0:01:18) ==========

```


### SciPy/NumPy/Python version and system information

```shell
1.14.0.dev0+1096.9c75b1f 1.26.4 sys.version_info(major=3, minor=12, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.12.3/lib/python3.12/site-packages/pythran
    version: 0.15.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.12.3/bin/python3.12
  version: '3.12'

```

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -3559,10 +3559,12 @@ class TestKurtosis(SkewKurtosisTest):
 @hypothesis.strategies.composite
 def ttest_data_axis_strategy(draw):
     # draw an array under shape and value constraints
-    dtype = npst.floating_dtypes()
     elements = dict(allow_nan=False, allow_infinity=False)
     shape = npst.array_shapes(min_dims=1, min_side=2)
-    data = draw(npst.arrays(dtype=dtype, elements=elements, shape=shape))
+    # The test that uses this, `test_pvalue_ci`, uses `float64` to test
+    # extreme `alpha`. It could be adjusted to test a dtype-dependent
+    # range of `alpha` if this strategy is needed to generate other floats.
+    data = draw(npst.arrays(dtype=np.float64, elements=elements, shape=shape))
 
     # determine axes over which nonzero variance can be computed accurately
     ok_axes = []
@@ -3711,8 +3713,7 @@ class TestStudentTest:
     def test_pvalue_ci(self, alpha, data_axis, alternative, xp):
         # test relationship between one-sided p-values and confidence intervals
         data, axis = data_axis
-        data = data.astype(np.float64, copy=True)  # ensure byte order
-        data = xp.asarray(data, dtype=xp.float64)
+        data = xp.asarray(data)
         res = stats.ttest_1samp(data, 0.,
                                 alternative=alternative, axis=axis)
         l, u = res.confidence_interval(confidence_level=alpha)
@@ -3722,6 +3723,7 @@ class TestStudentTest:
         res = stats.ttest_1samp(data, popmean, alternative=alternative, axis=axis)
         shape = list(data.shape)
         shape.pop(axis)
+        # `float64` is used to correspond with extreme range of `alpha`
         ref = xp.broadcast_to(xp.asarray(1-alpha, dtype=xp.float64), shape)
         xp_assert_close(res.pvalue, ref)
 

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -3711,8 +3711,8 @@ class TestStudentTest:
     def test_pvalue_ci(self, alpha, data_axis, alternative, xp):
         # test relationship between one-sided p-values and confidence intervals
         data, axis = data_axis
-        data = data.astype(copy=True)  # ensure byte order
-        data = xp.asarray(data)
+        data = data.astype(np.float64, copy=True)  # ensure byte order
+        data = xp.asarray(data, dtype=xp.float64)
         res = stats.ttest_1samp(data, 0.,
                                 alternative=alternative, axis=axis)
         l, u = res.confidence_interval(confidence_level=alpha)
@@ -3722,7 +3722,7 @@ class TestStudentTest:
         res = stats.ttest_1samp(data, popmean, alternative=alternative, axis=axis)
         shape = list(data.shape)
         shape.pop(axis)
-        ref = xp.broadcast_to(xp.asarray(1-alpha), shape)
+        ref = xp.broadcast_to(xp.asarray(1-alpha, dtype=xp.float64), shape)
         xp_assert_close(res.pvalue, ref)
 
 

